### PR TITLE
Python 3.7 Drop: deprecate copy_directory

### PR DESCRIPTION
**Pull request by vcidst**
**Proposed changes**:
- Deprecate copy_directory and use shutil.copytree() instead

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: tests/engine/storage/test_resource.py [hunk 1]
```diff
@@ -99,7 +100,9 @@ def test_resource_caching_if_already_restored_with_different_state(
     (temporary_directory / "another_file").touch()
 
     new_storage_dir = tmp_path_factory.mktemp("new dir")
-    rasa.utils.common.copy_directory(initial_storage_dir, new_storage_dir)
+    rasa.utils.common.copy_directory(
```
Comment (ancalita) [inline tests/engine/storage/test_resource.py:103]: This occurence of `copy_directory` was missed here, should be replaced also with `shutil.copytree`.

**Review by ancalita — APPROVED** [review]
LGTM 👍🏻 Only one missed occurrence of the custom method needs to be fixed.

File: tests/engine/storage/test_resource.py (hunk 1 above)
Comment (vcidst) [reply tests/engine/storage/test_resource.py]: ah yes, thank you!

File: rasa/utils/common.py [hunk 2]
```diff
@@ -516,37 +516,6 @@ def directory_size_in_mb(
...
-
-    # TODO: Drop this in favor of `shutil.copytree(..., dirs_exist_ok=True)` when
-    # dropping Python 3.7.
-
-    Args:
-        source: The directory whose contents should be copied to `destination`.
-        destination: The directory which should contain the content `source` in the end.
-
-    Raises:
-        ValueError: If destination is not empty.
```
Comment (vcidst) [inline rasa/utils/common.py:532]: @radovanZRasa The function here raises a `ValueError` if the destination is not empty. However `shutil.copytree(..., dirs_exist_ok=True)` doesn't raise if the destination is not empty. 

These are not identical behaviour. What should I do here?

**Review by radovanZRasa — APPROVED** [review]
LGTM

**Pull request merged**